services/venice-router/src/main/java/com/linkedin/venice/router/api/VeniceResponseDecompressor.java

```diff
@@ -42,30 +39,20 @@ public class VeniceResponseDecompressor {
 
   private final CompressionStrategy clientCompression;
   private final RouterStats<AggRouterHttpRequestStats> routerStats;
-  private final String storeName;
-  private final int version;
-  private final String kafkaTopic;
+  private final StoreVersionName storeVersionName;
   private final CompressorFactory compressorFactory;
 
   public VeniceResponseDecompressor(
-      boolean decompressOnClient,
+      CompressionStrategy clientCompression,
```

Comment:
Do we even need this?  clientCompression is a private final and I don't see it used outside this classes constructor.